src/sinks/elasticsearch/retry.rs

```diff
@@ -79,7 +113,31 @@ impl RetryLogic for ElasticsearchRetryLogic {
                 let body = String::from_utf8_lossy(response.http_response.body());
 
                 if body.contains("\"errors\":true") {
-                    RetryAction::DontRetry(get_error_reason(&body).into())
+                    match EsResultResponse::parse(&body) {
+                        Ok(resp) => {
+                            if self.retry_partial {
+                                if let Some((status, error)) =
+                                    resp.iter_status().find(|(status, _)| {
+                                        *status == StatusCode::TOO_MANY_REQUESTS
+                                            || status.is_server_error()
+                                    })
+                                {
+                                    let msg = if let Some(error) = error {
+                                        format!(
+                                            "partial error, status: {}, error type: {}, reason: {}",
+                                            status, error.err_type, error.reason
+                                        )
+                                    } else {
+                                        format!("partial error, status: {}", status)
+                                    };
```

Comment:
According to https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#bulk-api-response-body
>  error
    (object) Contains additional information about the failed operation.
    The parameter is only returned for failed operations.

It doesn't say it's always set for failed operations so I've added this condition just to be sure. Plus, even if we assume it's always set we will just replace this with an `.expect("Missing error message")` since this still needs to be an `Option` to be able to parse successful operations.